proxy/malformed_host_override.go

```diff
@@ -0,0 +1,122 @@
+// (C) Docker, Inc.
+// Redistributed under the Apache License 2.0
+
+package proxy
+
+import "net"
+
+// MalformedHostHeaderOverride is a wrapper to be able
+// to overcome the 400 Bad request coming from old docker
+// clients that send an invalid Host header.
```

Comment:
1. That sounds about right
2. Yes - this will happen as a by-product of running the tests with different docker versions